pkg/querier/queryrange/stats.go

```diff
@@ -91,39 +116,85 @@ func StatsCollectorMiddleware() queryrangebase.Middleware {
 			// collect stats and status
 			var statistics *stats.Result
 			var res promql_parser.Value
+			var queryType string
+			var totalEntries int
+
 			if resp != nil {
 				switch r := resp.(type) {
 				case *LokiResponse:
 					statistics = &r.Statistics
-					res = logqlmodel.Streams(r.Data.Result)
+					totalEntries = int(logqlmodel.Streams(r.Data.Result).Lines())
+					queryType = queryTypeLog
 				case *LokiPromResponse:
 					statistics = &r.Statistics
+					if r.Response != nil {
+						totalEntries = len(r.Response.Data.Result)
+					}
+					queryType = queryTypeMetric
+				case *LokiSeriesResponse:
+					statistics = &r.Statistics
+					totalEntries = len(r.Data)
+					queryType = queryTypeSeries
+				case *LokiLabelNamesResponse:
+					statistics = &r.Statistics
+					totalEntries = len(r.Data)
+					queryType = queryTypeLabel
 				default:
 					level.Warn(logger).Log("msg", fmt.Sprintf("cannot compute stats, unexpected type: %T", resp))
 				}
 			}
+
 			if statistics != nil {
 				// Re-calculate the summary: the queueTime result is already merged so should not be updated
 				// Log and record metrics for the current query
-				statistics.ComputeSummary(time.Since(start), 0)
+				statistics.ComputeSummary(time.Since(start), 0, totalEntries)
 				statistics.Log(level.Debug(logger))
 			}
 			ctxValue := ctx.Value(ctxKey)
 			if data, ok := ctxValue.(*queryData); ok {
 				data.recorded = true
 				data.statistics = statistics
 				data.result = res
+				data.queryType = queryType
 				p, errReq := paramsFromRequest(req)
 				if errReq != nil {
 					return nil, errReq
 				}
 				data.params = p
+
+				// Record information for metadata queries.
+				switch r := req.(type) {
+				case *LokiLabelNamesRequest:
+					data.label = getLabelNameFromLabelsQuery(r.Path)
+				case *LokiSeriesRequest:
+					data.match = r.Match
+				}
+
 			}
 			return resp, err
 		})
 	})
 }
 
+func getLabelNameFromLabelsQuery(path string) string {
+	if strings.HasSuffix(path, "/values") {
+
+		toks := strings.FieldsFunc(path, func(r rune) bool {
+			return r == '/'
+		})
+
+		// now assuming path has suffix `/values` label name should be second last to the suffix
+		// **if** there exists the second last.
```

Comment:
can we include a link to the docs for this endpoint? https://grafana.com/docs/loki/latest/api/#get-lokiapiv1labelnamevalues